### ISSUE TITLE
Skaffold panic when run with empty skaffold.yaml

### ISSUE THREAD

**Issue by rvrignaud** [GoogleContainerTools/skaffold#1216]
### Expected behavior
When run with an empty skaffold.yaml, I expect skaffold to output an error

### Actual behavior
Skaffold crashes for most command dev, run, fix.

```
skaffold run
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x1e02a57]

goroutine 1 [running]:
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.newRunner(0x2f732e0, 0xc00036c5d0, 0x22def20, 0xc0004da840, 0xc00036c5d0)
	/private/tmp/skaffold-20181027-61272-omcjia/src/github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd/runner.go:36 +0x67
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.run(0x22c3d40, 0xc00000c018, 0x0, 0x0)
	/private/tmp/skaffold-20181027-61272-omcjia/src/github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd/run.go:49 +0x9d
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.NewCmdRun.func1(0xc0004eec80, 0x2f91d40, 0x0, 0x0, 0x0, 0x0)
	/private/tmp/skaffold-20181027-61272-omcjia/src/github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd/run.go:34 +0x33
github.com/GoogleContainerTools/skaffold/vendor/github.com/spf13/cobra.(*Command).execute(0xc0004eec80, 0x2f91d40, 0x0, 0x0, 0xc0004eec80, 0x2f91d40)
	/private/tmp/skaffold-20181027-61272-omcjia/src/github.com/GoogleContainerTools/skaffold/vendor/github.com/spf13/cobra/command.go:762 +0x473
github.com/GoogleContainerTools/skaffold/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x2f5e060, 0xc0004eef00, 0xc0004eec80, 0xc0004eea00)
	/private/tmp/skaffold-20181027-61272-omcjia/src/github.com/GoogleContainerTools/skaffold/vendor/github.com/spf13/cobra/command.go:852 +0x2fd
github.com/GoogleContainerTools/skaffold/vendor/github.com/spf13/cobra.(*Command).Execute(0x2f5e060, 0xc00000c018, 0x22c3d40)
	/private/tmp/skaffold-20181027-61272-omcjia/src/github.com/GoogleContainerTools/skaffold/vendor/github.com/spf13/cobra/command.go:800 +0x2b
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app.Run(0x0, 0x0)
	/private/tmp/skaffold-20181027-61272-omcjia/src/github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/skaffold.go:27 +0x58
main.main()
	/private/tmp/skaffold-20181027-61272-omcjia/src/github.com/GoogleContainerTools/skaffold/cmd/skaffold/skaffold.go:26 +0x22
```

### Information

- Skaffold version: version : v0.17.0
- Operating system: macOS
- Contents of skaffold.yaml: empty file

### Steps to reproduce the behavior

1. touch skaffold.yaml
2. skaffold run

**Comment by aleksamagicka** [comment]
Same on Win 7 x64 Ultimate

**Comment by dgageot** [comment]
Fixed with #1221
